src/main/scala/mesosphere/marathon/core/storage/MarathonState.scala

```diff
@@ -0,0 +1,22 @@
+package mesosphere.marathon.core.storage
+
+import akka.util.ByteString
+import com.google.protobuf.{ MessageLite, Parser }
+import mesosphere.marathon.state.Timestamp
+
+trait MarathonState[Proto <: MessageLite] {
```

Comment:
I actually split the existing MarathonState in half, fromProto really belongs in the companion not the instance. This is actually used by the ZK typeclasses, marshaling is free, unmarshalling just requires the companion object to be specified.

And they do implement it already ;)